src/App.js

```diff
@@ -1,17 +1,201 @@
 import React from "react";
 import logo from "./logo.png";
 import "./App.css";
+import { WelcomeMsg } from "./WelcomeMsg.js";
+import { Instruction } from "./Instruction.js";
+import { InputForm } from "./InputForm.js";
+import { PlayForm } from "./PlayForm.js";
+import { LeaderBoard } from "./LeaderBoard.js";
+import { DiceImage } from "./DiceImage.js";
+import { rollDice, nextPlayer, nextRound } from "./utils.js";
+import { isLastPlayer, determineWinner, soundPlay } from "./utils.js";
+
+// define the audio clips used in this APP
+const audioClips = [
+  {
+    sound: "http://cd.textfiles.com/itcontinues/WIN/YTB22/RATTLE2.WAV",
+    label: "Dice Shake",
+  },
+  {
+    sound:
+      "https://soundbible.com/mp3/Short_triumphal_fanfare-John_Stracke-815794903.mp3",
+    label: "Victory sound",
+  },
+  {
+    sound:
+      "https://soundbible.com/mp3/jack_in_the_box-Mike_Koenig-710345321.mp3",
+    label: "Game music",
+  },
+];
 
 class App extends React.Component {
+  constructor(props) {
+    // Always call super with props in constructor to initialise parent class
+    super(props);
+    this.state = {
+      gameStarted: false,
+      numberOfPlayers: 2, // form input state
+      numberOfRounds: 3,
+      userRollDice: false,
+      userOrderDice: false,
+      userScores: [],
+      currentPlayer: 0,
+      currentRound: 0,
+      playerDiceRolls: [0, 0],
+      currentPlayerScore: 0,
+      diceOrder: 1,
+      audio: new Audio(audioClips[2].sound), // set background music
+      isMusicOn: false, // background music is initially OFF
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleContinue = this.handleContinue.bind(this);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({
+      [name]: parseInt(value),
+    });
+  }
+
+  handleSubmit(event) {
+    const { name, value } = event.target;
+    event.preventDefault();
+    this.state.audio.play(); // start playing background music
+    this.state.audio.loop = true;
```

Comment:
We never want to reassign state like so, but always use the setState function. By doing this here, you will run into trouble with missing rerenders and possibly out-of-sync state.